contrib/babelfishpg_tsql/src/pltsql_utils.c

```diff
@@ -1836,6 +1836,82 @@ check_fulltext_exist(const char *schema_name, const char *table_name)
 	return ft_index_name != NULL;
 }
 
+/*
+ * replace_special_chars_fts_impl
+ * Replace special characters in the input string with the unique hash
+ */
+char
+*replace_special_chars_fts_impl(char *input_str) {
+	size_t		input_len = strlen(input_str);
+	char		*replacement = NULL;
+	char		*result;
+	char		*unique_hashes[4];
+	const char	*special_chars[4] = {"~!&|@#$%^*+=\\;:<>?./", "`", "'", "_"};
+	StringInfoData	output_str;
+	
+	for (int i = 0; i < 4; i++) {
+		unique_hashes[i] = construct_unique_index_name("specialChars", psprintf("cat%d", i + 1));
+	}
+	
+	initStringInfo(&output_str);
+
+	for (size_t i = 0; i < input_len; i++) {
+		replacement = NULL;
+
+		for (int j = 0; j < 4; j++) {
+			if (strchr(special_chars[j], input_str[i]) != NULL) {
+				replacement = unique_hashes[j];
+				break;
+			}
+		}
+
+		if (replacement != NULL) {
+			/* Remove leading and trailing spaces around special characters */
+			size_t start = i;
+			size_t end = i;
+
+			/* Look for leading spaces */
+			while (start > 0 && isspace((unsigned char)input_str[start - 1])) {
+				start--;
+			}
+
+			/* Look for trailing spaces */
+			while (end < input_len - 1 && isspace((unsigned char)input_str[end + 1])) {
+				end++;
+			}
+
+			/* Copy the replacement with removed spaces */
+			appendStringInfo(&output_str, "-%s-" ,replacement);
+
+			/* Move the index to the end of the processed sequence */
+			i = end; 
+
+			/* skip multiple special characters in a row */
```

Comment:
We are throwing unsupported error for the search conditions right, but this replace_special_chars function is also applied on the column during the full-text index creation, so if we don't have this clause then let's say a phrase with consecutive special characters is inserted into the table, then it'll crash, so that's why added this clause.